backend/src/services/aws/ec2/ec2.service.ts

```diff
@@ -92,6 +91,12 @@ class EC2 {
 
     await this.setInstanceName(instanceId as string, name);
 
+    return {
+      instanceId: instanceId as string,
+    };
+  }
+
+  public async getPublicIpAddress(instanceId: string): Promise<string> {
```

Comment:
When you are naming/creating method always think about users of this method. This method has a **huge** side effect. It contains a blocking timer up to 90 seconds. It will definitely confuse other developers using this method. So you can either split this method into two: `waitUntillRunning` & `getPublicIpAddress` **or** rename to `waitUntilRunningAndReturnPublicIpAddress`